### PR TITLE
Document tagged addresses

### LINKED ISSUE #6989: Confused about doc in 'Service Definition' part
I'm reading the doc of consul .The doc gives a [configuration](https://www.consul.io/docs/agent/services.html#service-definition) file for the service, parts of it are as follows:
```
    "tagged_addresses": {
      "lan": {
        "address": "192.168.0.55",
        "port": 8000,
      },
      "wan": {
        "address": "198.18.0.23",
        "port": 80
      }
    },
    "port": 8000
```
So why the ports in tagged_addresses are different, does this mean that the wan address uses a proxy. Or the tagged_addresses configuration item have any other meaning?

### PR DESCRIPTION
**Pull request by blake**
Resolves #6989

### CODE REVIEW

**Review by freddygv — APPROVED** [review]
Looks accurate from my perspective

File: website/content/docs/discovery/services.mdx [hunk 1]
```diff
@@ -275,6 +275,255 @@ For Consul 0.9.3 and earlier you need to use `enableTagOverride`. Consul 1.0
...
+  }
+}
+```
+
+</CodeBlockConfig>
+</CodeTabs>
+
+#### Virtual
+
+Virtual tagged addresses are a logical address type which is used by
```
Comment (trujillo-adam) [inline website/content/docs/discovery/services.mdx:392]:
```suggestion
Virtual tagged addresses are logical address types used by
```

File: website/content/docs/discovery/services.mdx [hunk 2]
```diff
@@ -275,6 +275,255 @@ For Consul 0.9.3 and earlier you need to use `enableTagOverride`. Consul 1.0
...
+[Connect](/docs/connect) to route traffic to the associated destination service
+in the service mesh when:
```
Comment (trujillo-adam) [inline website/content/docs/discovery/services.mdx:394]: Just to be clear, both conditions should be met to route traffic--not either or, correct? If so, consider making it more explicit.

File: website/content/docs/discovery/services.mdx [hunk 3]
```diff
@@ -275,6 +275,255 @@ For Consul 0.9.3 and earlier you need to use `enableTagOverride`. Consul 1.0
...
+   [dialed directly](/docs/connect/config-entries/service-defaults#dialeddirectly).
+
+-> **Note**: Virtual addresses are not required to be routable IPs within the
+network. They are strictly a control plane construct used to provide a single
+address for the instances of a given logical service.
+
+##### Supported tags
+
+The following address tag can be configured for a service to advertise the
+logical address to be used to dial it over the service mesh.
```
Comment (trujillo-adam) [inline website/content/docs/discovery/services.mdx:407]:
```suggestion
Use the following address tag to specify the logical address the service should advertise when dialing over the service mesh.
```
I am not certain that I correctly restated the part about "dialing over the service mesh".

File: website/content/docs/discovery/services.mdx [hunk 4]
```diff
@@ -275,6 +275,255 @@ For Consul 0.9.3 and earlier you need to use `enableTagOverride`. Consul 1.0
...
+}
+```
+
+</CodeBlockConfig>
+</CodeTabs>
+
+#### WAN
+
+These addresses are intended to be accessible only when dialing a node or service
+from a remote WAN federated data center.
```
Comment (trujillo-adam) [inline website/content/docs/discovery/services.mdx:455]: It's probably just me, but I'm having trouble understanding the type of data center at line 455. 

Do we mean "remote WAN-federated data center", which would mean that the DC is remote and federated according to WAN addresses?

Do we mean "remote, WAN, federated data center", which means that the federated DC is also remote and identified by WAN address? 

Feel free to ignore this comment if it's just a gap in my understanding.

**Review by trujillo-adam — APPROVED** [review]
Looks good. I added a few minor suggestions and a few questions.

File: website/content/docs/discovery/services.mdx (hunk 2 above)
Comment (blake) [reply website/content/docs/discovery/services.mdx:394]: Yes, both conditions are required to be met. I can add such phrasing to the previous sentence.

File: website/content/docs/discovery/services.mdx (hunk 3 above)
Comment (blake) [reply website/content/docs/discovery/services.mdx:407]:
```suggestion
Use the following address tag to specify the logical address at which the service can be reached by other services in the mesh.
```
Is this perhaps more clear?

File: website/content/docs/discovery/services.mdx (hunk 4 above)
Comment (blake) [reply website/content/docs/discovery/services.mdx:455]: Consul has a feature called [translated addresses](https://www.consul.io/api-docs#translated-addresses) which allow the catalog to return different addresses in the discovery response based on whether the requestor is in the same DC, or a different DC that is connected via WAN federation.

The docs for the [`translate_wan_addrs`](https://www.consul.io/docs/agent/options#translate_wan_addrs) config option offers a better explanation on this behavior.

> If set to true, Consul will prefer a node's configured WAN address when servicing DNS and HTTP requests for a node in a remote datacenter. This allows the node to be reached within its own datacenter using its local address, and reached from other datacenters using its WAN address, which is useful in hybrid setups with mixed networks. This is disabled by default.

These `wan` tagged addresses are the same as the `WAN address` mentioned in the previous paragraph.

File: website/content/docs/discovery/services.mdx (hunk 3 above)
Comment (trujillo-adam) [reply website/content/docs/discovery/services.mdx:412]: Yes, I think I was thrown off by the term 'dial' -- thinking it meant something more technical than literally just a service communicating with another another service

File: website/content/docs/discovery/services.mdx (hunk 4 above)
Comment (trujillo-adam) [reply website/content/docs/discovery/services.mdx:460]: OK, then I _believe_ the correct statement would be: 

"Define a WAN address when services need to remotely access a federated datacenter at a WAN address." 

Yeah?

**Review by trujillo-adam — APPROVED** [review]
Made a minor suggestion based on your explanation

File: website/content/docs/discovery/services.mdx (hunk 4 above)
Comment (blake) [reply website/content/docs/discovery/services.mdx:460]: We settled on the following phrasing.
> Define a WAN address for the service or node when it should be accessed at an alternate address by services in a remote datacenter.

**Comment by hc-github-team-consul-core** [comment]
:cherries: If backport labels were added before merging, cherry-picking will start automatically.

To retroactively trigger a backport after merging, add backport labels and re-run https://circleci.com/gh/hashicorp/consul/427248.

**Comment by hc-github-team-consul-core** [comment]
:cherries::white_check_mark: Cherry pick of commit 8a396ae73f56c1a4f48c1b6a272a241e3c7be863 onto `stable-website` succeeded!

**Comment by hc-github-team-consul-core** [comment]
:cherries::white_check_mark: Cherry pick of commit 8a396ae73f56c1a4f48c1b6a272a241e3c7be863 onto `release/1.10.x` succeeded!

**Pull request merged**